youtube_dl/extractor/srgssr.py

```diff
@@ -33,102 +50,167 @@ def _get_tokenized_src(self, url, video_id, format_id):
-            'url': image['url'],
-        } for image in media_data.get('Image', {}).get('ImageRepresentations', {}).get('ImageRepresentation', [])]
+        subtitles = self.extract_subtitles(media_data, bu, media_type)
 
-        preference = qualities(['LQ', 'MQ', 'SD', 'HQ', 'HD'])
         formats = []
-        for source in media_data.get('Playlists', {}).get('Playlist', []) + media_data.get('Downloads', {}).get('Download', []):
-            protocol = source.get('@protocol')
-            for asset in source['url']:
-                asset_url = asset['text']
-                quality = asset['@quality']
-                format_id = '%s-%s' % (protocol, quality)
-                if protocol.startswith('HTTP-HDS') or protocol.startswith('HTTP-HLS'):
-                    asset_url = self._get_tokenized_src(asset_url, media_id, format_id)
-                    if protocol.startswith('HTTP-HDS'):
-                        formats.extend(self._extract_f4m_formats(
-                            asset_url + ('?' if '?' not in asset_url else '&') + 'hdcore=3.4.0',
-                            media_id, f4m_id=format_id, fatal=False))
-                    elif protocol.startswith('HTTP-HLS'):
-                        formats.extend(self._extract_m3u8_formats(
-                            asset_url, media_id, 'mp4', 'm3u8_native',
-                            m3u8_id=format_id, fatal=False))
+        preference = qualities(['LQ', 'MQ', 'SD', 'HQ', 'HD'])
+        for source in media_data.get('resourceList', []):
+            protocol = source.get('protocol')
+            quality = source.get('quality')
+            encoding = source.get('encoding')
+            mime_type = source.get('mimeType')
+            format_url = source.get('url')
+            format_id = '%s-%s-%s' % (protocol, encoding, quality)
+
+            if protocol in ('HDS', 'HLS'):
+                asset_url = self._get_tokenized_src(
+                    format_url, media_id, format_id)
+                if protocol == 'HDS':
+                    formats.extend(self._extract_akamai_formats(
+                        asset_url, media_id))
                 else:
+                    formats.extend(self._extract_m3u8_formats(
+                        asset_url, media_id, 'mp4', 'm3u8_native',
+                        m3u8_id=format_id, fatal=False
+                    ))
+            elif protocol in ('HTTP', 'HTTPS', 'RTMP'):
+                formats.append({
+                    'format_id': format_id,
+                    'ext': mimetype2ext(mime_type) if mime_type else None,
+                    'url': format_url,
+                    'preference': preference(quality)
+                })
+            podcast_keys = ('podcastSdUrl', 'podcastHdUrl')
+            podcast_qualities = ('SD', 'HD')
+
+        # This is needed because for audio medias the podcast url is usually
+        # always included, even if is only an audio segment and not the
+        # whole episode.
```

Comment:
do you have an example for this case(whole podcast URL present for an audio segment)?